scripts/parser-tests/typescript/error-codes.js

```diff
@@ -0,0 +1,71 @@
+/*
+This file includes the code of TS Diagnostics error that should also be thrown from babel-parser.
+The TypeScript parser is highly tolerant on errors so these error would not produce parseDiagnostics which can be checked
+in the parser test runner. We check these error codes against the stderr log in the build/typescript/tests/baselines/reference
+
+Note that babel-parser should not throw for the TypeChecking Diagnostics
+
+The commented out diagnostic codes will introduce false positive cases that should be addressed in separate PRs.
+*/
+
+module.exports = [
+  // "TS1005", // '{0}' expected.
+  "TS1009", // Trailing comma not allowed.
```

Comment:
TypeScript does not push `parseDiagnostics` for all and not limited to these errors. 🤷